.github/workflows/ci.yml

```diff
@@ -120,3 +120,60 @@ jobs:
         curl -L https://github.com/ndmitchell/hlint/releases/download/v${HLINT_VERSION}/hlint-${HLINT_VERSION}-x86_64-linux.tar.gz --output hlint.tar.gz
         tar -xvf hlint.tar.gz
         ./hlint-${HLINT_VERSION}/hlint src/ test/
+
+  stan:
+    name: stan
+    runs-on: ubuntu-latest
+    env:
+      CABAL_VERSION: "3.6"
+      # `stan` Github Release supports this version as of now. If we want to run stan on multiple
+      # GHC versions, we need to build `stan` from source.
+      GHC_VERSION: "8.10.1"
+      STAN_VERSION: "0.0.1.0"
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - uses: haskell/actions/setup@v2.0
+      id: setup-haskell-cabal
+      name: Setup Haskell
+      with:
+        ghc-version: ${{ env.GHC_VERSION }}
+        cabal-version: ${{ env.CABAL_VERSION }}
+
+    - name: Configure
+      run: |
+        cabal configure --enable-tests --enable-benchmarks --enable-documentation --test-show-details=direct --write-ghc-environment-files=always
+
+    - name: Freeze
+      run: |
+        cabal freeze
+
+    - uses: actions/cache@v3
+      name: Cache ~/.cabal/store
+      with:
+        path: ${{ steps.setup-haskell-cabal.outputs.cabal-store }}
+        key: ${{ runner.os }}-${{ env.GHC_VERSION }}-${{ hashFiles('cabal.project.freeze') }}
+
+    - name: Install dependencies
+      run: |
+        cabal build all --only-dependencies
```

Comment:
I copied all of this from the cabal job of this workflow. But I don't know if we really need it, I'm unfamiliar with all of these `cabal` commands. If there is no need for them, I can remove them.

Reply:
Yeah, I'm afraid we need it 😞 
`stan` needs the project to be built with the same GHC version as `stan` itself. Since we don't build with GHC 8.10.7, we need to repeat all of this.

Maybe we can use GHC 8.10.7 here but then we'll need to build `stan` from sources instead of simply downloading it. So, not much better.

Let's leave it like this for now 👍🏻